### PR TITLE
Fix/condo/doma 3800/major ticket export upgrade

### PR DESCRIPTION
**Pull request by AntonAL**
Added logging into `tasks` module before and after scheduling of a task in real mode

Job progress reporting to Bull on "exportTickets" task
This will prevent Bull to move the job to "failed" state with error "job stalled more than allowable limit" when it exceeds its timeout (30 seconds)

👏 Displaying real progress for TicketExportTask based on total records to process

Added testing of chunk-by-chunk execution for `exportTickets` worker function

### CODE REVIEW

File: apps/condo/domains/ticket/tasks/exportTicketsTask.js [hunk 1]
```diff
@@ -2,11 +2,11 @@ const get = require('lodash/get')
 const isNull = require('lodash/isNull')
 const map = require('lodash/map')
 const compact = require('lodash/compact')
-const { TicketExportTask, TicketStatus } = require('../utils/serverSchema')
+const { TicketExportTask, TicketStatus, Ticket } = require('../utils/serverSchema')
```
Comment (sitozzz) [inline apps/condo/domains/ticket/tasks/exportTicketsTask.js:5]: Absolute import

File: apps/condo/domains/ticket/tasks/exportTicketsTask.js [hunk 2]
```diff
@@ -2,11 +2,11 @@ const get = require('lodash/get')
...
 const { exportRecords } = require('@condo/domains/common/utils/serverSchema/export')
 const { createTask } = require('@condo/keystone/tasks')
 const { getSchemaCtx } = require('@condo/keystone/schema')
-const { loadTicketsBatchForExcelExport, loadTicketCommentsForExcelExport, loadClassifiersForExcelExport } = require('../utils/serverSchema')
+const { buildTicketsLoader, loadTicketCommentsForExcelExport, loadClassifiersForExcelExport } = require('../utils/serverSchema')
```
Comment (sitozzz) [inline apps/condo/domains/ticket/tasks/exportTicketsTask.js:9]: Absolute import plz

File: apps/condo/domains/ticket/tasks/exportTicketsTask.spec.js [hunk 3]
```diff
@@ -10,19 +10,26 @@ const {
...
 const { createTestTicket } = require('@condo/domains/ticket/utils/testSchema')
-const { exportTickets } = require('./exportTicketsTask')
+const { exportTicketsWorker } = require('./exportTicketsTask')
 const { createTestTicketExportTask, TicketExportTask } = require('../utils/testSchema')
 const { PROCESSING, EXCEL } = require('@condo/domains/common/constants/export')
 
 const index = require('../../../index')
 const { COMPLETED } = require('@condo/domains/common/constants/export')
 const { catchErrorFrom } = require('@condo/domains/common/utils/testSchema')
+const { EXPORT_PROCESSING_BATCH_SIZE } = require('../../common/constants/export')
```
Comment (sitozzz) [inline apps/condo/domains/ticket/tasks/exportTicketsTask.spec.js:20]: Absolute import

File: packages/keystone/tasks.js [hunk 4]
```diff
@@ -60,8 +60,9 @@ function removeCronTask (name, cron, opts = {}) {
 }
 
 async function _scheduleRemoteTask (name, preparedArgs, preparedOpts) {
+    logger.info(`Scheduling task "${name}" with args "${JSON.stringify(preparedOpts)}"`)
     const job = await taskQueue.add(name, { args: preparedArgs }, preparedOpts)
-
+    logger.info(`Task "${name}" with args "${JSON.stringify(preparedOpts)}" was scheduled and assigned to Bull job with id "${job.id}"`)
```
[the remark this answers is not in the document]
Comment (AntonAL) [reply packages/keystone/tasks.js:65]: It's not the same message, logging before and after adding to queue helps to determine successfull result of adding job operation.

File: apps/condo/domains/ticket/tasks/exportTicketsTask.spec.js [hunk 5]
```diff
@@ -10,19 +10,26 @@ const {
...
+
+const jobMock = {
+    progress: jest.fn().mockImplementation(),
+}
 
 describe('exportTicketsTask', () => {
     describe('exportTickets', () => {
         setFakeClientMode(index)
+        // We need more time to generate tickets to process them chunk-by-chunk
+        jest.setTimeout(60000)
```
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/ticket/tasks/exportTicketsTask.spec.js:22]: 15 seconds can be not enough as it was set in `nodejs.condo.test.yml`

File: packages/keystone/tasks.js [hunk 6]
```diff
@@ -72,22 +73,40 @@ async function _scheduleRemoteTask (name, preparedArgs, preparedOpts) {
...
 
+/**
+ * Because Bull does not participates in case of execution tasks with `FAKE_WORKER_MODE`,
+ * an instance of a job, that is passed to worker function should have the same
+ * methods as in Bull.
+ * This helps to avoid handling `FAKE_WORKER_MODE` in every custom worker function.
+ * Ideally the interface should be exactly the same as in "Bull", but currently only `.progress`
+ * method is used in worker functions. So, make it simple.
+ */
+class InProcessJob {
```
Comment (pahaz) [inline packages/keystone/tasks.js:84]:
```suggestion
class InProcessFakeJob {
```

File: packages/keystone/tasks.js [hunk 7]
```diff
@@ -60,8 +60,9 @@ function removeCronTask (name, cron, opts = {}) {
 }
 
 async function _scheduleRemoteTask (name, preparedArgs, preparedOpts) {
+    logger.info({ msg: 'Scheduling task', name, preparedArgs, preparedOpts, FAKE_WORKER_MODE })
```
Comment (pahaz) [inline packages/keystone/tasks.js:63]:
```suggestion
    logger.info({ msg: 'Scheduling task', name, data: { args: preparedArgs, opts: preparedOpts } })
```

File: packages/keystone/tasks.js [hunk 8]
```diff
@@ -60,8 +60,9 @@ function removeCronTask (name, cron, opts = {}) {
...
     const job = await taskQueue.add(name, { args: preparedArgs }, preparedOpts)
-
+    logger.info({ msg: 'Created job for task', name, job, preparedArgs, preparedOpts, FAKE_WORKER_MODE })
```
Comment (pahaz) [inline packages/keystone/tasks.js:65]: you really don't need to log this

**Review by sitozzz — APPROVED** [review]

**Pull request merged**